crates/services/txpool_v2/src/tests/mock_db.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ports::{
+    AtomicView,
+    TxPoolPersistentStorage,
+};
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    entities::{
+        coins::coin::{
+            Coin,
+            CompressedCoin,
+        },
+        relayer::message::Message,
+    },
+    fuel_tx::{
+        BlobId,
+        Contract,
+        ContractId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+    fuel_vm::BlobBytes,
+};
+use std::{
+    collections::{
+        HashMap,
+        HashSet,
+    },
+    sync::{
+        Arc,
+        Mutex,
+    },
+};
+
+#[derive(Default)]
+pub struct Data {
+    pub coins: HashMap<UtxoId, CompressedCoin>,
+    pub contracts: HashMap<ContractId, Contract>,
+    pub blobs: HashMap<BlobId, BlobBytes>,
+    pub messages: HashMap<Nonce, Message>,
+    pub spent_messages: HashSet<Nonce>,
+}
+
+#[derive(Clone, Default)]
+pub struct MockDb {
+    pub data: Arc<Mutex<Data>>,
+}
+
+impl MockDb {
+    pub fn insert_coin(&self, coin: Coin) {
+        self.data
+            .lock()
+            .unwrap()
+            .coins
+            .insert(coin.utxo_id, coin.compress());
+    }
+
+    pub fn insert_dummy_blob(&self, blob_id: BlobId) {
+        self.data
+            .lock()
+            .unwrap()
+            .blobs
+            .insert(blob_id, vec![123; 123].into());
+    }
+
+    pub fn insert_message(&self, message: Message) {
+        self.data
+            .lock()
+            .unwrap()
+            .messages
+            .insert(*message.id(), message);
+    }
+
+    pub fn spend_message(&self, id: Nonce) {
+        self.data.lock().unwrap().spent_messages.insert(id);
+    }
+}
+
+impl TxPoolPersistentStorage for MockDb {
+    fn utxo(&self, utxo_id: &UtxoId) -> StorageResult<Option<CompressedCoin>> {
+        Ok(self.data.lock().unwrap().coins.get(utxo_id).cloned())
+    }
+
+    fn contract_exist(&self, contract_id: &ContractId) -> StorageResult<bool> {
+        Ok(self
+            .data
+            .lock()
+            .unwrap()
+            .contracts
+            .contains_key(contract_id))
+    }
+
+    fn blob_exist(&self, blob_id: &BlobId) -> StorageResult<bool> {
+        Ok(self.data.lock().unwrap().blobs.contains_key(blob_id))
+    }
+
+    fn message(&self, id: &Nonce) -> StorageResult<Option<Message>> {
+        Ok(self.data.lock().unwrap().messages.get(id).cloned())
+    }
+}
+
+pub struct MockDBProvider(pub MockDb);
+
+impl AtomicView for MockDBProvider {
```

Comment:
Do we still need to impl this since we aren't taking a dep on `AtomicView` anymore?

Reply:
We are still taking a persistent storage that can us an atomic view that herself implement `TxPoolDb` I just changed the fact that now the `AtomicView` is defined in the ports of this module